FitGoal/FitGoal/Singup&Singin/GreetingViewController.swift

```diff
@@ -0,0 +1,167 @@
+            color: #colorLiteral(red: 0.36, green: 0.37, blue: 0.4, alpha: 1),
+            kern: -0.12
+        )
+        label.attributedText = text
+        return label
+    }()
+    
+    private let createAccountButton: UIButton = {
+        let button = UIButton()
+        button.backgroundColor = #colorLiteral(red: 0.5647058824, green: 0.07450980392, blue: 0.9568627451, alpha: 1)
+        let title = "Create Account".formattedText(font: "Roboto-Bold", size: 17, color: .white, kern: 0)
+        button.setAttributedTitle(title, for: .normal)
+        return button
+    }()
+    
+    private let greetingText: UITextView = {
+        let text = UITextView()
+        let string = "Start transforming the way \n you enjoy you life"
+        let attributedString = string.formattedText(
+            font: "Roboto-Light",
+            size: 15,
+            color: UIColor(red: 0.52, green: 0.53, blue: 0.57, alpha: 1),
+            kern: 0,
+            lineSpacing: 6
+        )
+        text.isEditable = false
+        text.isScrollEnabled = false
+        text.isSelectable = false
+        text.backgroundColor = .clear
+        text.attributedText = attributedString
+        text.textAlignment = .center
+        return text
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = #colorLiteral(red: 0.9647058824, green: 0.9647058824, blue: 0.9647058824, alpha: 1)
+        let views = [
+            backgroundView,
+            avatarView,
+            mainLabel,
+            socialMediaAuthentication,
+            createAccountButton,
+            greetingText,
+            linkToSignUp
+        ]
+        view.addMultipleSubviews(views)
+        setConstraints()
+        
+        let tap = UITapGestureRecognizer(target: self, action: #selector(presentViewController(_:)))
+        createAccountButton.addGestureRecognizer(tap)
+        
+        linkToSignUp.delegate = self
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        backgroundView.frame = CGRect(
+            x: 0,
+            y: 0,
+            width: view.bounds.width,
+            height: 1/3 * view.bounds.height
+        )
+        
+        socialMediaAuthentication.frame = CGRect(
+            x: 0,
+            y: 2/3 * view.bounds.maxY,
+            width: view.bounds.width,
+            height: 1/3 * view.bounds.height
+        )
+        
+        createAccountButton.frame = CGRect(
+            x: 16,
+            y: view.bounds.midY + 50,
+            width: view.bounds.width - 32,
+            height: 52
+        )
+        
+        avatarView.frame = CGRect(
+            x: view.bounds.midX - 42,
+            y: 130,
+            width: 104,
+            height: 104
+        )
+        
+        createAccountButton.layer.cornerRadius = createAccountButton.bounds.height/2
+    }
+
+    @objc private func presentViewController(_ sender: UITapGestureRecognizer) {
```

Comment:
After you add this function as a target you don't need to consider the state inside the function.